src/Polly.Core/README.md

```diff
@@ -168,96 +162,130 @@ var ResiliencePipeline = new ResiliencePipelineBuilder()
-- **Events**: These are invoked when significant events occur within the resilience strategy.
-- **Generators**: These are used when the resilience strategy requires a certain value from the caller.
+## Resilience Strategy Delegates
 
-## Delegate Signature Guidelines
+Individual resilience strategies make use of several delegate types:
 
-The suggested signatures for these delegates are as follows:
+- **Predicates**: Vital for determining whether a resilience strategy should handle the given execution result.
+- **Events**: Triggered when significant actions or states occur within the resilience strategy.
+- **Generators**: Invoked when the resilience strategy needs specific information or values from the caller.
+
+Recommended signatures for these delegates are:
 
 **Predicates**
-- `Func<OutcomeArguments<T, TArgs>, ValueTask<bool>>`: This is the predicate for the generic outcome.
-- `Func<OutcomeArguments<object, TArgs>, ValueTask<bool>>`: This is the predicate for the non-generic outcome.
+
+- `Func<Args<TResult>, ValueTask<bool>>` (Reactive)
 
 **Events**
-- `Func<OutcomeArguments<T, TArgs>, ValueTask>`: This is the event for the generic outcome.
-- `Func<OutcomeArguments<object, TArgs>, ValueTask>`: This is the event for the non-generic outcome.
-- `Func<Args, ValueTask>`: This is the event utilized by strategies that do not operate with an outcome (for example, Timeout, RateLimiter).
+
+- `Func<Args<TResult>, ValueTask>` (Reactive)
+- `Func<Args, ValueTask>` (Non-Reactive)
 
 **Generators**
-- `Func<OutcomeArguments<T, TArgs>, ValueTask<TValue>>`: This is the generator for the generic outcome.
-- `Func<OutcomeArguments<object, TArgs>, ValueTask<TValue>>`: This is the generator for the non-generic outcome.
-- `Func<Args, ValueTask<TValue>>`: This is the generator used by strategies that do not operate with an outcome (for example, Timeout, RateLimiter).
 
-It's essential to note that all these delegates are asynchronous and return a `ValueTask`. 
+- `Func<Args<TResult>, ValueTask<TValue>>` (Reactive)
+- `Func<Args, ValueTask<TValue>>` (Non-Reactive)
+
+
+These delegates accept either `Args` or `Args<TResult>` arguments, which encapsulate event information. Note that all these delegates are asynchronous and return a `ValueTask`.
+
+> [!NOTE]
+> When setting up delegates, consider using the `ResilienceContext.ContinueOnCapturedContext` property if your user code interacts with a synchronization context (as in asynchronous UI applications like Windows Forms or WPF).
+
+### Delegate Arguments
+
+For non-reactive strategies, the `Args` structure might resemble:
+
+```csharp
+public readonly struct OnTimeoutArguments
+{
+    public OnTimeoutArguments(ResilienceContext context, TimeSpan timeout)
+    {
+        Context = context;
+        Timeout = timeout;
+    }
+
+    public ResilienceContext Context { get; } // Always include the Context property
```

Comment:
Rather than needing a comment, feels like we should have some sort of type constraint so that it has to be implemented. I know we used to have something and it's been removed, but it seems weird that if this is needed we have to ask someone to provide it like duck-typing, rather than some sort of interface or constraint.

Reply:
I don't have  any idea how to do it except introducing something like `IResilienceArguments` again.

I have changed the wording, none of these are a "must", it's just a recommendation.